master/internal/root.go

```diff
@@ -0,0 +1,36 @@
+package internal
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+// root manages the lifecycle of all actors in the Determined master and
+// defines a supervision strategy specifically for the master.
+func root(ctx *actor.Context) error {
+	switch msg := ctx.Message().(type) {
+	case actor.PreStart, actor.PostStop:
+		return nil
+	case actor.ChildFailed:
+		switch msg.Child.Address() {
+		case sproto.K8sRMAddr, sproto.AgentRMAddr:
+			return errors.Wrapf(msg.Error, "resource mananger crashed, exiting")
+		case sproto.PodsAddr, sproto.AgentsAddr:
+			return errors.New("pods or agents manager crashed, exiting")
```

Comment:
Maybe wrap here as well for consistency? Though, actually, right now the error is going to show up three times (once when the actor fails initially, once when the root actor fails, and finally from the new `log.withError` in `core.go`), so it might look nicer to skip the wrap entirely instead.